project/Dependencies.scala

```diff
@@ -21,7 +22,7 @@ object Dependencies {
   lazy val embeddedRedis = "it.ozimov"            % "embedded-redis"      % Versions.embeddedRedis
   lazy val redis4cats = "dev.profunktor"         %% "redis4cats-effects"  % Versions.redis4cats
   lazy val redis4catsLog4cats = "dev.profunktor" %% "redis4cats-log4cats" % Versions.redis4cats
-  lazy val scaffeine = "com.github.blemale"      %% "scaffeine"           % Versions.scaffeine
+  lazy val scaffeine = ("com.github.blemale"     %% "scaffeine"           % Versions.scaffeine).cross(CrossVersion.for3Use2_13)
```

Comment:
I ported scaffeine some time ago, see https://github.com/blemale/scaffeine/pull/211 – version 5.1.0 is cross-build with Scala 3.

Apart from that, the PR looks good to me.

Reply:
scaffeine 5.x supports jdk11+ only.
we crossbuild against jdk8.
could you apply that patch to 4.x and ask to release 4.2.0?

Reply:
@ybasket I created a PR to scaffeine. If it's not accepted, I suggest keep using 2.13 version in compact mode.

Reply:
Be careful with using the 2.13 artefact, it would bring in `scala-collection-compat` and `scala-java8-compat` in their 2.13 versions – that can clash at runtime if they're present in their 3.0 versions as well. So I would at least suggest to test a demo app which has them included to see whether you get a fatal Java exception like `NoSuchMethodError`.

See the warning in the sbt docs: https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Scala+3+specific+cross-versions